integration-cli/docker_api_containers_test.go

```diff
@@ -1947,3 +1947,46 @@ func (s *DockerSuite) TestContainerKillCustomStopSignal(c *check.C) {
 	err = waitInspect(id, "{{.State.Running}} {{.State.Restarting}}", "false false", 30*time.Second)
 	c.Assert(err, checker.IsNil)
 }
+
+// Regression test for #34307
+// Make sure that a container shares its namespace with ID.
+// Test renaming a container which namespace is shared by another container.
+func (s *DockerSuite) TestContainerShareNamespaceByName(c *check.C) {
+	// Namespace is not supported on Windows
+	testRequires(c, DaemonIsLinux, SameHostDaemon, NotUserNamespace)
+
+	name := "base"
+	out := runSleepingContainer(c, "--name", name)
+	id := strings.TrimSpace(out)
+	expected := "container:" + id
+
+	dockerCmd(c, "run", "--name=share", "--ipc", "container:"+name, "--pid", "container:"+name, "--net", "container:"+name, "busybox")
```

Comment:
diff looks good.

Yes, a fake `containerGetter` really (not the whole daemon). It shouldn't be hard to implement, it's something like this:

```go

type fakeContainerGetter struct {
    containers map[string]*container.Container
}

func (f *fakeContainerGetter) GetContainer(cid string) (*container.Container, error) {
    container, ok := f.containers[cid]
    if !ok {
         return nil, errors.New("container not found")
    }
    return container, nil
}
```

A test like `TestAdjustCPUShares` would also be ok.